sc/src/main/javascript/bundles/experiment-page/src/TSnePlotViewRoute.js

```diff
@@ -27,15 +31,32 @@ const TSnePlotViewRoute = (props) => {
                     speciesName={species}
                     experimentAccession={experimentAccession}
                     ks={ks}
-                    selectedK={Number(search.k) || props.selectedK || props.ks[0]}
+                    metadata={metadata}
+                    selectedColourBy={search.k || search.metadata || props.selectedK || props.ks[0].toString()}
+                    selectedColourByCategory={search.colourBy || 'clusters'} // Is the plot coloured by clusters or metadata
                     highlightClusters={search.clusterId ? JSON.parse(search.clusterId) : []}
                     perplexities={perplexities}
                     selectedPerplexity={Number(search.perplexity) || props.perplexities[Math.round((perplexities.length - 1) / 2)]}
                     geneId={search.geneId || ``}
                     height={800}
                     onSelectGeneId={ (geneId) => { updateUrlSearch({ name: `geneId`, value: geneId }) } }
-                    onChangeK={ (k) => { updateUrlSearch({ name: `k`, value: k }) } }
-                    onChangePerplexity={ (perplexity) => { updateUrlSearch({ name: `perplexity`, value: perplexity }) } }
+                    onChangePerplexity={ (perplexity) => { updateUrlSearch({ name: `perplexity`, value: perplexity }) }}
+                    onChangeColourBy={
+                      (colourByCategory, colourByValue) => {
+                        const query = new URLSearchParams(history.location.search)
```

Comment:
Interesting! I’ve been using URI.js for such a long time that I have no idea of the state of official APIs. I’ll have a look at it a bit more in-depth.